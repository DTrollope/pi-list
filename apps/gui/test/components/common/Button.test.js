import React from 'react';
import { shallow } from 'enzyme';
import Button from 'components/common/Button';
import renderer from 'react-test-renderer';

describe('components/common/Button', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Button label="Test button" />);
    });

    describe('Button Component', () => {
        it('should include a custom className', () => {
            wrapper.setProps({ className: 'custom-classname' });

            expect(wrapper.find('.lst-btn').hasClass('custom-classname')).toBe(true);
        });

        it('should call the prop `onClick` when the user click in the button', () => {
            const onClickMock = jest.fn();
            wrapper.setProps({ onClick: onClickMock });

            expect(onClickMock).not.toBeCalled();

            wrapper.find('.lst-btn').simulate('click');

            expect(onClickMock).toBeCalled();
        });
    });

    describe('Button Snapshots', () => {
        it('should render a default button with the correct label', () => {
            const tree = renderer
                .create(<Button label="test"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a success button', () => {
            const tree = renderer
                .create(<Button type="success" label="test"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a warning button', () => {
            const tree = renderer
                .create(<Button type="warning" label="test"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a info button', () => {
            const tree = renderer
                .create(<Button type="info" label="test"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it(`should render a default button if the button type isn't defined`, () => {
            const tree = renderer
                .create(<Button type="lol" label="test"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a default outlined button', () => {
            const tree = renderer
                .create(<Button label="test" outlined/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a primary outlined button', () => {
            const tree = renderer
                .create(<Button type="primary" label="test" outlined/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a success outlined button', () => {
            const tree = renderer
                .create(<Button type="success" label="test" outlined/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a danger outlined button', () => {
            const tree = renderer
                .create(<Button type="danger" label="test" outlined/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a warning outlined button', () => {
            const tree = renderer
                .create(<Button type="warning" label="test" outlined/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a info outlined button', () => {
            const tree = renderer
                .create(<Button type="info" label="test" outlined/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a button without margin', () => {
            const tree = renderer
                .create(<Button type="info" label="test" noMargin/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a disabled button', () => {
            const tree = renderer
                .create(<Button label="test" disabled/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a button as a link', () => {
            const tree = renderer
                .create(<Button label="test" link />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a button with an icon', () => {
            const tree = renderer
                .create(<Button label="test" icon="delete" />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
