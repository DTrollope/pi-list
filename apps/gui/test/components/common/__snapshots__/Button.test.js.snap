// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Button Button Snapshots should render a button as a link 1`] = `
<button
  className="lst-btn lst-btn--link"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a button with an icon 1`] = `
<button
  className="lst-btn"
  disabled={false}
  onClick={null}
  type="button"
>
  <i
    className="lst-icons"
  >
    delete
  </i>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a button without margin 1`] = `
<button
  className="lst-btn lst-btn--info lst-btn--no-margin"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a danger outlined button 1`] = `
<button
  className="lst-btn lst-btn--danger"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a default button if the button type isn't defined 1`] = `
<button
  className="lst-btn"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a default button with the correct label 1`] = `
<button
  className="lst-btn"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a default outlined button 1`] = `
<button
  className="lst-btn"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a disabled button 1`] = `
<button
  className="lst-btn"
  disabled={true}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a info button 1`] = `
<button
  className="lst-btn lst-btn--info"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a info outlined button 1`] = `
<button
  className="lst-btn lst-btn--info"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a primary outlined button 1`] = `
<button
  className="lst-btn lst-btn--primary"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a success button 1`] = `
<button
  className="lst-btn lst-btn--success"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a success outlined button 1`] = `
<button
  className="lst-btn lst-btn--success"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a warning button 1`] = `
<button
  className="lst-btn lst-btn--warning"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;

exports[`components/common/Button Button Snapshots should render a warning outlined button 1`] = `
<button
  className="lst-btn lst-btn--warning"
  disabled={false}
  onClick={null}
  type="button"
>
  test
</button>
`;
